### DISCUSSION TITLE
dlopen error when my library dlopen another library in Android 9

### DISCUSSION THREAD

**Discussion by FranzKafkaYu** [android/ndk#1683] [Q&A] [answered]
now I encountered an error when my library(**A.so, which is locate in / system/lib64) dlopen another library(**B.so, which is in /system/ etc/automotive/),and i got the dlerror: dlopen failed , library **B.so needed or dlopened by **A.so is not accessible for the namespace " default".

And I tried modifying /system/etc/public.libraries.txt,add **A.so or **B.so but it doesn't work.here is what i can find something referenced here in official link:https://source.android.com/devices/architecture/vndk/linker-namespace

Does anyone can help me now ?

I just don't know,why dlopen a third-party library in a specific path can be restricted.

**Comment by enh-google** [comment, marked as answer]
you probably want the android-building googlegroup, since this is a question about building the OS, not about the NDK.
